src/bsoncxx/oid.cpp

```diff
@@ -23,12 +23,7 @@
 namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 
-const oid::init_tag_t oid::init_tag{};
-
-oid::oid() : _is_valid(false) {
-}
-
-oid::oid(init_tag_t) : _is_valid(true) {
```

Comment:
Ditto re inability to comment where needed, but note that oid_compare gets a lot simpler if you drop the _is_valid stuff.